workspaces/arborist/lib/edge.js

```diff
@@ -179,6 +179,11 @@ class Edge {
 
   get spec () {
     if (this.overrides?.value && this.overrides.value !== '*' && this.overrides.name === this.#name) {
+      // If this edge has the same overrides field as the source, then we're not applying an override for this edge.
+      if (this.overrides === this.#from.overrides) {
```

Comment:
☝️ Anywhere there is a `this.#from` it may be `undefined`. We should reference it as `this.#from?.overrides` and in a follow up fix the rest of the `#from` use.

Reply:
Below this in the code there is
```js
// from the real root, not the virtual one
const pkg = this.#from.sourceReference
  ? this.#from.sourceReference.root.package
  : this.#from.root.package
```
which is also not safe as `.root` in both cases may be `null`. So it needs `?.root` (separate pr though)